lib/engine.rb

```diff
@@ -1,64 +1,130 @@
 # frozen_string_literal: true
 
-# Class for game state
 class Game
```

Comment:
[OPTIONAL]
Maybe try putting this class in a separate file as it is better separation of code.
If you really want to have it in this engine file you can always require it